File: 2024/day1 - dart/pt2.dart
import 'dart:io';
void main(){
  var total = 0;
  final leftList =<int>[];
  final rightList =<int>[];
  
  final doc = File('input.txt').readAsLinesSync();
  for (var line in doc){
    final cleanLine = line.split("   ");
    final lListNum = int.parse(cleanLine[0]);
    final rListNum = int.parse(cleanLine[1]);
    leftList.add(lListNum);
    rightList.add(rListNum);
  }
  for (var lNum in leftList){
    var multiplier = 0;
    for (var rNum in rightList){
      if(rNum == lNum){
        multiplier +=1;
      }
    }
    total += lNum*multiplier;
  }
  print(total);
}

File: 2024/day1/pt2.py
leftList = []
rightList = []
totalSimScore = 0
with open('input.txt', 'r') as locations:
    for lines in locations:
        twoLines = lines.replace('\n',"").split('   ')
        leftList.append(int(twoLines[0]))
        rightList.append(int(twoLines[1]))

for lNum in leftList:
    simMulti = 0
    for rNum in rightList:
        if rNum == lNum:
            simMulti+=1
    totalSimScore += simMulti*lNum
print(totalSimScore)